t/t5799-gvfs-helper.sh

```diff
@@ -0,0 +1,361 @@
+#!/bin/sh
+
+test_description='test gvfs-helper and GVFS Protocol'
+
+. ./test-lib.sh
+
+# Set the port for t/helper/test-gvfs-protocol.exe from either the
+# environment or from the test number of this shell script.
+#
+test_set_port GIT_TEST_GVFS_PROTOCOL_PORT
+
+# Setup the following repos:
+#
+#    repo_src:
+#        A normal, no-magic, fully-populated clone of something.
+#        No GVFS (aka VFS4G).  No Scalar.  No partial-clone.
+#        This will be used by "t/helper/test-gvfs-protocol.exe"
+#        to serve objects.
+#
+#    repo_t1:
+#        An empty repo with no contents nor commits.  That is,
+#        everything is missing.  For the tests based on this repo,
+#        we don't care why it is missing objects (or if we could
+#        actually use it).  We are only testing explicit object
+#        fetching using gvfs-helper.exe in isolation.
+#
+REPO_SRC="$PWD"/repo_src
+REPO_T1="$PWD"/repo_t1
```

Comment:
I'll leave this note open until the cleanup.  I've been burned too many times because of the whitespace in the "trash directory" parent directory name....